pyicloud/base.py

```diff
@@ -28,6 +32,57 @@
 logger = logging.getLogger(__name__)
 
 
+class PyiCloudPasswordFilter(logging.Filter):
+    def __init__(self, password):
+        self.password = password
+
+    def filter(self, record):
+        message = record.getMessage()
+        if self.password in message:
+            record.msg = message.replace(self.password, "*" * 8)
+            record.args = []
+
+        return True
+
+
+class PyiCloudSession(requests.Session):
+    def __init__(self):
+        super(PyiCloudSession, self).__init__()
+
+    def request(self, *args, **kwargs):
+
+        # Charge logging to the right service endpoint
+        callee = inspect.stack()[2]
+        module = inspect.getmodule(callee[0])
+        logger = logging.getLogger(module.__name__)
+
+        logger.debug("%s %s %s", args[0], args[1], kwargs.get('data', ''))
+
+        response = super(PyiCloudSession, self).request(*args, **kwargs)
+
+        json = None
+        if 'application/json' in response.headers['Content-Type']:
+            json = response.json()
+            logger.debug(json)
+
+        reason = json.get('errorMessage') or json.get('reason')
+        if not reason and isinstance(json.get('error'), str):
```

Comment:
I'm pretty confident `json.loads` returns unicode objects for key and value names.  Using six, you can check this in a Python2/3 compatible way by doing something like --

```python
from six import string_types
isinstance(json.get('error'), string_types)
```